src/gui/TrackContainerView.cpp

```diff
@@ -377,18 +377,26 @@ void TrackContainerView::dropEvent( QDropEvent * _de )
 		//it->toggledInstrumentTrackButton( true );
 		_de->accept();
 	}
-	else if( type == "samplefile" || type == "pluginpresetfile" 
-		|| type == "soundfontfile" || type == "vstpluginfile"
-		|| type == "patchfile" )
+	else if (type == "samplefile" || type == "pluginpresetfile"
+			|| type == "soundfontfile" || type == "vstpluginfile"
+			|| type == "patchfile")
 	{
-		InstrumentTrack * it = dynamic_cast<InstrumentTrack *>(
-				Track::create( Track::InstrumentTrack,
-								m_tc ) );
-		PluginFactory::PluginInfoAndKey piakn =
-			pluginFactory->pluginSupportingExtension(FileItem::extension(value));
-		Instrument * i = it->loadInstrument(piakn.info.name(), &piakn.key);
-		i->loadFile( value );
-		//it->toggledInstrumentTrackButton( true );
+		const TrackView * tv = trackViewAt(_de->pos().y() - timeLineWidgetHeight());
+		if (type == "samplefile" && tv && tv->getTrack()->type() == Track::SampleTrack)
+		{
+			_de->accept();
+		}
```

Comment:
I think you don't even need this part, `SampleTrackView::dragEnterEvent` will do the job. If this works, you need to change only `SampleTrackView`.

Reply:
Seems to work. :+1: